docs/source/installation/guide_complete/index.rst

```diff
@@ -0,0 +1,363 @@
+.. _installation:guide-complete:
+
+===========================
+Complete installation guide
+===========================
+
+The :ref:`quick installation guide <installation:guide-quick>` is designed to make the setup as simple and portable as possible.
+However, the resulting setup has some :ref:`limitations <installation:guide-quick:limitations>` concerning the available functionality and performance.
+This guide provides detailed information and instructions to set up a feature-complete and performant installation.
+
+To get a working installation of AiiDA, the following steps are required:
+
+#. :ref:`Install the Python Package <installation:guide-complete:python-package>`
+#. :ref:`Create a profile <installation:guide-complete:create-profile>`
+
+
+.. _installation:guide-complete:python-package:
+
+Install the Python Package
+======================
+
+.. important::
+    AiiDA requires a recent version of Python.
+    Please refer to the `Python Package Index (PyPI) <https://pypi.org/project/aiida-core/>`_ for the minimum required version.
+
+To install AiiDA, the ``aiida-core`` Python package needs to be installed which can be done in a number of ways:
+
+
+.. tab-set::
+
+    .. tab-item:: pip
+
+        Installing ``aiida-core`` from PyPI.
+
+        #. Install `pip <https://pip.pypa.io/en/stable/installation/>`_.
+        #. Install ``aiida-core``:
+
+           .. code-block:: console
+
+               pip install aiida-core
+
+    .. tab-item:: conda
+
+        Installing ``aiida-core`` using Conda.
+
+        #. Install `conda <https://conda.io/projects/conda/en/latest/user-guide/install/index.html>`_.
+
+        #. Create an environment and install ``aiida-core``:
+
+           .. code-block:: console
+
+               conda create -n aiida -c conda-forge aiida-core
+
+    .. tab-item:: source
+
+        Installing ``aiida-core`` directory from source.
+
+        #. Install `git <https://www.git-scm.com/book/en/v2/Getting-Started-Installing-Git>`_
+        #. Clone the repository from Github
+
+           .. code-block:: console
+
+               git clone https://github.com/aiidateam/aiida-core
+
+        #. Install `pip <https://pip.pypa.io/en/stable/installation/>`_.
+        #. Install ``aiida-core``:
+
+           .. code-block:: console
+
+               cd aiida-core
+               pip install -e .
+
+           The ``-e`` flag installs the package in editable mode which is recommended for development.
+           Any changes made to the source files are automatically picked up by the installation.
+
+
+.. _installation:guide-complete:python-package:optional-requirements:
+
+Optional requirements
```

Comment:
I understand why this is here in the section "Install Python Package", though, I propose we move that to the end, or fold it away. Most of these additional requirements (possibly apart from `atomic_tools` and `notebook`) are for `dev`s and not relevant for the average user. We could mention those and make the full list foldable to reduce distractions?

Reply:
> Most of these additional requirements (possibly apart from atomic_tools and notebook) are for devs and not relevant for the average user.

Are you sure? To me, only 3 out of 8 are just for development (`docs`, `pre-commit`, `tests`).

Reply:
I see your point. I agree that it can be superfluous information for most cases, which is why I didn't include it at all in the quick install guide. However, here we are in the _complete_ guide, so I think it is justified to provide this information. And although flow is important, I don't think it is that disruptive that we should move it elsewhere, where it would almost always be more disconnected from the most logical place. But as mentioned elsewhere, I would prefer to discuss this in person with everybody else, so we can hash all these suggestions out.

Reply:
Yeah, I agree. Overall, I'm also fine with leaving it there. Let's put it on the agenda of the upcoming aiidateam meeting. Should be the last, more involved blocking PR that we should all be happy with 👍🏽